sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/util/IntervalUtils.scala

```diff
@@ -774,19 +747,32 @@ object IntervalUtils {
         UNIT_SUFFIX,
         UNIT_END = Value
   }
-  private final val intervalStr = UTF8String.fromString("interval")
-  private def unitToUtf8(unit: IntervalUnit): UTF8String = {
-    UTF8String.fromString(unit.toString)
+  private final val intervalStr = unitToUtf8("interval")
+  private def unitToUtf8(unit: String): UTF8String = {
+    UTF8String.fromString(unit)
+  }
+  private final val yearStr = unitToUtf8("year")
+  private final val monthStr = unitToUtf8("month")
+  private final val weekStr = unitToUtf8("week")
+  private final val dayStr = unitToUtf8("day")
+  private final val hourStr = unitToUtf8("hour")
+  private final val minuteStr = unitToUtf8("minute")
+  private final val secondStr = unitToUtf8("second")
+  private final val millisStr = unitToUtf8("millisecond")
+  private final val microsStr = unitToUtf8("microsecond")
+  private final val nanosStr = unitToUtf8("nanosecond")
+
+  def intervalUnits: Seq[UTF8String] = {
+    Seq(yearStr,
+      monthStr,
+      weekStr,
+      dayStr,
+      hourStr,
+      minuteStr,
+      secondStr,
+      millisStr,
+      microsStr)
```

Comment:
Is it used somewhere except of tests? Just wonder why did you move it from `ExpressionParserSuite`?

Reply:
> Is it used somewhere except of tests? Just wonder why did you move it from `ExpressionParserSuite`?

Move back, how about current?